src/test/test-specifier.c

```diff
@@ -127,13 +127,18 @@ TEST(specifier_real_path_missing_file) {
 }
 
 TEST(specifiers) {
+        int r;
+
         for (const Specifier *s = specifier_table; s->specifier; s++) {
                 char spec[3];
                 _cleanup_free_ char *resolved = NULL;
 
                 xsprintf(spec, "%%%c", s->specifier);
 
-                assert_se(specifier_printf(spec, SIZE_MAX, specifier_table, NULL, NULL, &resolved) >= 0);
+                r = specifier_printf(spec, SIZE_MAX, specifier_table, NULL, NULL, &resolved);
+                if (r == -ENOENT) /* machine-id might be missing in build chroots */
+                        continue;
```

Comment:
Maybe, it is better to check if the specifier is actually for machine ID, and machine ID doesn't exist.

Reply:
Also.